quit/core.py

```diff
@@ -415,14 +415,44 @@ def getFileReferenceAndContext(self, blob, commit):
         return self._blobs.get(blob)
 
     def applyQueryOnCommit(self, parsedQuery, parent_commit_ref, target_ref, query=None,
-                           default_graph=[], named_graph=[]):
+                           default_graph=[], named_graph=[], queryType=None, comment=None):
         """Apply an update query on the graph and the git repository."""
         graph, commitid = self.instance(parent_commit_ref)
         resultingChanges, exception = graph.update(parsedQuery)
         if exception:
             # TODO need to revert or invalidate the graph at this point.
             pass
-        oid = self.commit(graph, resultingChanges, 'New Commit from QuitStore', parent_commit_ref,
+
+        graphuri = None
+        print(resultingChanges)
+        print("comment: ")
+        print(comment)
+
+        if comment is not None:
+            queryType = comment
+        elif len(resultingChanges) > 1:
+            queryType = 'Edit resource in'
+        for entry in resultingChanges:
+            if "delta" in entry:
+                for x in entry["delta"]:
+                    graphuri = str(x)
+                    if queryType == 'Modify':
+                        ls = entry["delta"][x]
+                        if len(ls) == 1 and "removals" in ls[0]:
+                            queryType = 'Remove resource in'
+                        elif len(ls) == 1 and "additions" in ls[0]:
+                            queryType = 'Add resource in'
+
+        if queryType is not None and graphuri is not None:
+            if queryType == 'InsertData' or queryType == 'Load':
+                message = 'Insert data into Graph <' + graphuri + '>'
+            elif queryType == 'DeleteData' or queryType == 'DeleteWhere':
+                message = 'Delete data from Graph <' + graphuri + '>'
+            else:
+                message = queryType + ' Graph <' + graphuri + '>'
+        else:
+            message = 'New Commit from QuitStore'
+        oid = self.commit(graph, resultingChanges, message, parent_commit_ref,
```

Comment:
Yes, `'New Commit from QuitStore'` is a placeholder. It should be a message given by the client not generated based on the query, as the query is already represented in the commit message.

Reply:
See https://github.com/AKSW/QuitStore/issues/237

Reply:
Thanks for the tip! Yes I understand, just want to show the possibility during demonstration =)